src/components/scenegraph/Toolbar.js

```diff
@@ -69,13 +82,42 @@ export default class Toolbar extends React.Component {
    * Try to write changes with aframe-inspector-watcher.
    */
   writeChanges = () => {
-    const xhr = new XMLHttpRequest();
-    xhr.open('POST', 'http://localhost:51234/save');
-    xhr.onerror = () => {
-      alert('aframe-watcher not running. This feature requires a companion service running locally. npm install aframe-watcher to save changes back to file. Read more at supermedium.com/aframe-watcher');
-    };
-    xhr.setRequestHeader('Content-Type', 'application/json');
-    xhr.send(JSON.stringify(AFRAME.INSPECTOR.history.updates));
+    const baseUrl = "http://localhost:8888/";
+    const apiEndpointScene = AFRAME.scenes[0].getAttribute("id").replace("-scene", "");
+    const apiEndpoint = "api/admin/v1/scene/";
+    const getUrl = baseUrl + apiEndpoint + apiEndpointScene;
+    let objects = [];
+    let objectChanges = [];
+    axios.get(getUrl, {
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+        .catch((error) => {
+            throw error;
+        })
+        .then(function (response) {
+          objects = response.data.objects;
+          for(var id in AFRAME.INSPECTOR.history.updates)
+            objectChanges.push([parseInt(id.replace("-obj", "")), AFRAME.INSPECTOR.history.updates[id]]);
+          objectChanges.sort();
+          let i = 0;
+          let j = 0;
+          while(i < objects.length && j < objectChanges.length){
```

Comment:
Instead of a double for loop here with the if statement, would it be helpful to have the objects `response.data.objects` be a map instead? I can make that change on the backend if that would be helpful.

Reply:
At the moment this is O( max(len(objects), len(objectChanges) ), using the map would reduce it to O( len(objectChanges) ), I don't think it would be that much of a difference